likecoin/assets/js/likecoin.js

```diff
@@ -79,7 +114,29 @@ async function handleUpdateId(newId) {
   updateStatus.textContent = await res.text();
 }
 
-loginBtn.addEventListener('click', login);
-updateBtn.addEventListener('click', () => handleUpdateId());
+loginBtn.addEventListener('click', onLoginClick);
+changeBtn.addEventListener('click', onChangeClick);
+
+async function fetchLikeCoinID(newAddress) {
+  try {
+    const res = await fetch(`${challengeUrl}?wallet=${newAddress}`);
+    await res.json();
+    address = newAddress;
+    showError('.needLogin');
+  } catch (e) {
+    showError('.needLikeCoinId');
+  }
+}
+
+async function likecoinInit() {
+  const newAddress = await pollForWebThree();
+  if (address !== newAddress && newAddress) {
+    await fetchLikeCoinID(newAddress);
+  }
+}
 
 likecoinInit();
+setInterval(async () => {
+  likecoinInit(); // loop for web3 changes
```

Comment:
This may execute the async functions concurrently, will it cause race condition?
Will the code below be better?
```
const looper = async () => {
  await likecoinInit(); // may need try-catch block?
  setTimeout(looper, 3000);
});
looper();

Reply:
Or simply
```
(async () => {
  while (true) {
    await likecoinInit(); // may need try-catch block?
    await sleep(3000);
  }
})();